framework/src/mesh/MooseMesh.C

```diff
@@ -879,6 +879,33 @@ MooseMesh::buildElemIDInfo()
   comm().min(_min_ids);
 }
 
+std::map<dof_id_type, std::set<dof_id_type>>
+MooseMesh::getElemIDMapping(const std::string & from_id_name, const std::string & to_id_name) const
+{
+  auto & mesh_base = getMesh();
+
+  if (!mesh_base.has_elem_integer(from_id_name))
+    mooseError("Mesh does not have an element integer names as ", from_id_name);
+  if (!mesh_base.has_elem_integer(to_id_name))
+    mooseError("Mesh does not have an element integer names as ", to_id_name);
+
+  auto id1 = mesh_base.get_elem_integer_index(from_id_name);
+  auto id2 = mesh_base.get_elem_integer_index(to_id_name);
+
+  std::map<dof_id_type, std::set<dof_id_type>> id_map;
+  auto ids = getAllElemIDs(id1);
+  for (const auto & id : ids)
+    id_map[id] = std::set<dof_id_type>();
```

Comment:
```suggestion
    id_map[id] = std::unordered_set<dof_id_type>();
```

Reply:
Can we keep this as `set`? I think it should not be significantly more costly.

Reply:
Do you have a reason? Less costly would be a reason. Not significantly more costly is not a reason. If I could buy a carton of eggs for $5 I would not buy the identical eggs next to it that cost $6.

Reply:
I do not have a particular reason at this moment. When we do fuel shuffling, we move one assembly to another assembly. We assume the two assemblies have the exact same mesh thus shuffling won't change the mesh. Also we assume the two assemblies have the same number of depletion zones in the same order, thus we can set up a map with this function (called with assembly id name and depletion zone id name) from one depletion zone to another and use the map to shuffle the atomic densities of depletion zones. It is possible safer to assume depletion zone ids in an assembly is ordered thus using set here is preferred otherwise different ways of visiting the assembly elements may result into different orderings. So I prefer to have set here. This shuffling capability has not been in place yet.

Reply:
This function is returning a value. It is very likely the caller will just call this function and later discard the map after some operations.

Reply:
> This function is returning a value. It is very likely the caller will just call this function and later discard the map after some operations.

I can discard a map, but while I'm using it I might prefer to have O(1) lookups instead of O(log(N) lookups. But if you think you want want ordering in your usage, then that is a good reason for the performance penalty